control_toolbox/src/pid_ros.cpp

```diff
@@ -241,21 +244,35 @@ bool PidROS::initialize_from_ros_parameters()
     set_parameter_event_callback();
   }
 
-  if (antiwindup_strat_str == "none")
+  RCLCPP_WARN_EXPRESSION(
+    node_logging_->get_logger(), antiwindup_strat_str == "legacy",
+    "Using the legacy anti-windup technique is deprecated. This option will be removed by the ROS "
+    "2 Kilted Kaiju release.");
+
+  AntiWindupStrategy antiwindup_strat;
+  antiwindup_strat.set_type(antiwindup_strat_str);
+  antiwindup_strat.i_max = i_max;
+  antiwindup_strat.i_min = i_min;
+  antiwindup_strat.tracking_time_constant = tracking_time_constant;
+  antiwindup_strat.legacy_antiwindup = antiwindup;
+  antiwindup_strat.error_deadband = error_deadband;
+
+  try
   {
-    std::cout << "Old anti-windup technique is deprecated. "
-                 "This option will be removed by the ROS 2 Kilted Kaiju release."
-              << std::endl;
+    antiwindup_strat.validate();
+  }
+  catch (const std::exception & e)
+  {
+    RCLCPP_ERROR(node_logging_->get_logger(), "Invalid antiwindup strategy: %s", e.what());
+    return false;
   }
```

Comment:
can't we skip this check here, as it will be done anyways in 
```
bool Pid::set_gains(
  double p, double i, double d, double u_max, double u_min,
  const AntiWindupStrategy & antiwindup_strat)
```

Reply:
Done